src/ZEO/asyncio/client.py

```diff
@@ -534,15 +580,14 @@ def register_failed(self, protocol, exc):
 
     verify_result = None  # for tests
 
-    @future_generator
+    @coroutine
     def verify(self, server_tid):
-        """cache verification and invalidation."""
-        self.verify_invalidation_queue = []  # See comment in init :(
-
+        """cache verification and invalidation -- run as task."""
         protocol = self.protocol
+        call = protocol.server_call
         try:
             if server_tid is None:
```

Comment:
We have dropped ZEO 4 support. Therefore, we know that `server_tid` is not `None`.

Reply:
True. I tried to remove this quickly but this turns out to also require adjusting many tests in ZEO/asyncio/tests.py because there we `respond` to register with None return. I agree we should fix it all, but could we please get this work reviewed and merged first. It already has more than 100 patches and it took a lot of effort on my side to reach current state. I've created https://github.com/zopefoundation/ZEO/issues/221 so that this does not get forgotten.